### DISCUSSION TITLE
Flow hijacks the latest content that is placed on the clipboard, while the clipboard shows the "previous" content

### DISCUSSION THREAD

**Discussion by raindropsfromsky** [Flow-Launcher/Flow.Launcher#1291] [Q&A] [answered]
Dear all,

I am facing this issue in 1.10.0 alpha very often.

When I place anything on the clipboard (text/image) with a copy/cut operation, Flow hijacks that latest content.
The clipboard continues to have the content that was placed in it previously.
So, if I paste the content anywhere, the _previous_ content is pasted, _not_ the latest.  

If I exit Flow, it hands off its hijacked content to the clipboard and exits.
Now if I paste the content anywhere, the _latest_ content is pasted.
I do not have to repeat the copy/cut operation.

From this "black box" testing, I gather the following impression:
This is as if the clipboard is now behaving like a stack instead of a single placeholder.
Further, the top of the stack is hijacked by Flow often (but not always).
So the top of the stack is like a private variable within the Flow code.
When I exit Flow, it releases this private variable and makes it public before exiting.
Thus, the latest content now sits on the top of the clipboard stack.
Now the clipboard behaves as expected.

There is something in the code that makes this branching random instead of consistent.

Reporting for Windows Home 10, 64 bits.

I have already reported this as an issue, but there is no progress there.
Meanwhile, the problem is bugging me.
So I have posted that here as a discussion.

**Comment by taooceros** [comment, marked as answer]
Have you installed the clipboard plugin?

**Reply by raindropsfromsky** [reply]
I have installed the **clipboard history** plugin, and it is turned **on**.
But I am not using it at all.
(To be precise, I have never used the designated keyword "cb" while using Flow.)
Come to think of it, I am not clear how to use it, or what it is supposed to do.
I thought it actually cycles through the clipboard history with _UpArrow_ and _DownArrow_.
But it does not work that way.

Anyhow, I am seeing the effect _outside_ Flow, and it is apparently _random_.
(That is to say, there may be a pattern, but I haven't found it yet.)

I tried the following experiment: 
I placed some text on the clipboard and then used Flow to search-
(a) some other random text and 
(b) the text on the clipboard (I used CTRL+V in Flow) 
In both cases, the clipboard content is NOT hijacked by Flow.

Thus I am unable to replicate the condition consistently.

**Reply by taooceros** [reply]
I believe that's the issue? You can try uninstall it. Maybe when it is monitoring the clipboard, it somehow prevent writing.

**Reply by raindropsfromsky** [reply]
I wanted to uninstall the plugin, but there is no control for that!
All I can do is to turn it off.

The most logical place for the "uninstall" button is in the "Plugins" page, and also the "plugin store" page.
In fact, I am not sure why there are two separate pages for plugins. They can be easily combined!

Visually, the _uninstall_ button can be a _trash can_ icon or a simple _X_ button.

BTW this may be a wordaround in my case, but a permanent solution needs to be found.
Also, how do I use this plugin?? There is no help text on that!

**Reply by taooceros** [reply]
Type pm uninstall clipboard.

**Reply by taooceros** [reply]
I agree with you that we may need a gui for uninstall.

**Reply by taooceros** [reply]
I don't remember how exactly that plugin works, but somehow it just record the clipboard information as history. Just like the win+v i guess. So when you type the actionkeyword, you should be able to see the history of your clipboard.
